File: cpp/program/Author.cpp
#pragma once
#include <bits/stdc++.h>

using namespace std;

// kelas author
class Author
{
private:
    // atribut-atribut private
    string name;
    string nationality;

public:
    // konstruktor kosong
    Author()
    {
    }

    // konstruktor dengan parameter untuk mengisi atribut
    Author(string name, string nationality)
    {
        this->name = name;
        this->nationality = nationality;
    }

    // setter name
    void setName(string name)
    {
        this->name = name;
    }

    // getter name
    string getName()
    {
        return name;
    }

    // setter nationality
    void setNationality(int nationality)
    {
        this->nationality = nationality;
    }

    // getter nationality
    string getNationality()
    {
        return nationality;
    }

    ~Author()
    {
    }
};

File: java/program/Author.java
class Author {
    // atribut-atribut private
    private String name;
    private String nationality;

    // constructor dengan parameter
    Author(String name, String nationality) {
        this.name = name;
        this.nationality = nationality;
    }

    // getter dan setter untuk name
    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    // getter dan setter untuk nationality
    public void setNationality(String nationality) {
        this.nationality = nationality;
    }

    public String getNationality() {
        return nationality;
    }
}
